### PR TITLE
Add ResumableJS Typings

### PR DESCRIPTION
**Pull request by DanielMcAssey**
case 1. Add a new type definition.
- [ Y ] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [ Y ] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [ Y ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

New Typings for ResumableJS

### CODE REVIEW

File: resumablejs/resumable.d.ts [hunk 1]
```diff
@@ -0,0 +1,374 @@
+// Type definitions for Resumable.js v1.0.2
```
Comment (mhegazy) [inline resumablejs/resumable.d.ts]: filename should match the package name. so it should be `resumablejs/resumablejs.d.ts`
see http://definitelytyped.org/guides/contributing.html#naming-the-file

File: resumablejs/resumable.d.ts [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+	 
+	 /**
+	 * Drop event handler
+	 **/
+	handleDropEvent(e: Event): void;
+	
+    /**
+     * Listen for event from Resumable.js (see below)
+     **/
+    on(event: string, callback: Function): void;
```
Comment (mhegazy) [inline resumablejs/resumable.d.ts]: the most general overload should be last. Typescript resolves signatures in order. with this setup this is the overload that will always be picked.

File: resumablejs/resumable.d.ts [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+     * Returns the total size of the upload in bytes.
+     **/
+    getSize(): number;
+    getOpt(o: string): any;
+    
+    // Events
+	/**
+	 * Change event handler
+	 **/
+	handleChangeEvent(e: Event): void;
```
Comment (mhegazy) [inline resumablejs/resumable.d.ts]: please fix indentation. also applies to the rest of the file.

File: resumablejs/resumable.d.ts [hunk 4]
```diff
@@ -0,0 +1,374 @@
+// Type definitions for Resumable.js v1.0.2
+// Project: https://github.com/23/resumable.js
+// Definitions by: Daniel McAssey <https://github.com/DanielMcAssey>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace Resumable {
```
Comment (mhegazy) [inline resumablejs/resumable.d.ts]: I am asuming that Resumable is going to be only used as an npm pacakge, is this correct?

if so, then you do not need to put this name in the global namespace. Instead define these inside the module declaration, e.g.:

```ts

declare module 'resumablejs' {
   namespace Resumable {
    interface ConfigurationHash { ... }
    class Resumable { ... }
  }
  export = Resumable;
}
```

**Comment by DanielMcAssey** [comment]
Those have now been resolved.

**Comment by mhegazy** [comment]
thanks!

**Pull request merged**